pkg/validation/app.go

```diff
@@ -536,3 +535,26 @@ func contains(s []string, e string) bool {
 	}
 	return false
 }
+
+func isClusterSingleton(entry v1alpha1.AppCatalogEntry, cr, app v1alpha1.App) bool {
+	if entry.Spec.Restrictions.ClusterSingleton {
+		if key.IsInOrgNamespace(cr) &&
```

Comment:
I'd rather we get !key.IsInOrgNamespace(cr) first and that you use an else if to make it easier to read

Reply:
Also, if we can check this with key.IsInOrgNamespace(cr) && key.ClusterID(cr) == key.ClusterID(app) instead of the kubeconfig, I'd rather we do it :)

Reply:
That way you could have 

```go
if !key.IsInOrgNamespace(cr) {
  return true
} else if key.ClusterID(cr) == key.ClusterID(app)  { // or cr.Spec.KubeConfig.Context.Name == app.Spec.KubeConfig.Context.Name pick you solution
  return true
}
return false

```

Reply:
you can even simplify with 

```go
if !key.IsInOrgNamespace(cr) {
  return true
}
return key.ClusterID(cr) == key.ClusterID(app) // or cr.Spec.KubeConfig.Context.Name == app.Spec.KubeConfig.Context.Name
```